codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -209,22 +314,31 @@ public void incrementWindowSize(Http2Stream stream, int delta) throws Http2Excep
         assert ctx == null || ctx.executor().inEventLoop();
         if (stream.id() == CONNECTION_STREAM_ID) {
             // Update the connection window
-            connectionState().incrementStreamWindow(delta);
+            final AbstractState connectionState = connectionState();
+            connectionState.incrementStreamWindow(delta);
+
+            // Check if we need to notify listener of writability changes
+            if (incrementWindowSizeWritabilityVisitor != null && connectionState.isWritabilityChanged()) {
```

Comment:
@Scottmitch I wonder if this isn't really a special case.  Given that all streams are arranged in a tree, a stream isn't really `writable` unless its parent is `writable`.  I wonder if this logic should be generalized for all streams that have children.  WDYT?